projects/core/src/event/tms.event.ts

```diff
@@ -0,0 +1,26 @@
+import { Observable } from 'rxjs';
+import { createFrom } from '../util/create-from';
+import { EventService } from './event.service';
+
+export class TmsEvent {
+  static type = 'TmsEvent';
+  event: string;
+  payload: any;
+}
+
+export abstract class AbstractTmsEventCollector {
```

Comment:
As it's abstract by definition, we don't have to say it in the name IMO.  so would go with just `TmsEventCollector`.